### ISSUE TITLE
Question(s), Features request(s), and (maybe?) bugs so far.

### ISSUE THREAD

**Issue by Remski01** [Zrips/CMI#146]
1, Add an option to send all players an mail by using sendall tag. (aka /mail sendall <message>)

2, Ability to remove the "cmi" and just ket it be as exampe /sethome instead /cmi sethome

3, **Maybe more** things after my second attempt with testing the plugin.

Questions:

1, Can i convert all EssX data to CMI data? This include there money and home's.

2, **Maybe more** things after my second attempt with testing the plugin.

Please note that this is my first "close up" in what the plugin can do.

**Comment by Zrips** [comment]
1. Will keep in mind this one
2. You can enable automatic alias in alias.yml file and then you will have that /sethome command or create your own with aliaseditor

1. You can import essentials data with /cmi importfrom essentials home warp nick logoutlocation money
and if you have worth.yml you can simply copy/paste it as CMI uses same format

**Comment by DutchUser** [comment]
**1, Can you add an option to use an reason after /afk** 
**as example:** /afk i'm going eat. and you then see this message in the afk message, like "<player> is now afk, reason <text behind /afk>" 

**Command examples:**
/afk - Just for whitout reason. "Player is now AFK." [cmi.afk]
/afk <text/I'm going eat(example)> - "Player is now AFK, reason <text>" [cmi.afk.message]
(Admin) /afk <player> <his reason> - "Player is now AFK, reason <text>" [cmi.afk.admin]
(Admin) /afk <player> "Player is now AFK." [cmi.afk.admin]

2, How the hell can i not send myself an mail?
It wil just say what for options i have with the /mail command.

**Comment by DutchUser** [comment]
Can you please add the next commands please,

/craft - Open an craftingtable
/tpcancel <player> - Cancel the lastest teleport request or any from an x player.

**Comment by DutchUser** [comment]
**Hi. I have 2 questions.**

**First question:** 
If i remove EssX and place CMI, will nobody lose there inventory?

**Second question:**
How can i set that an kit can only be claimed 1 time?

I'm on last part where we replace EssX with CMI on our main server.

**Comment by MCTyler** [comment]
@Zrips Has said that this is not a forum, so I suggest opening a New issue for each issue you are requesting. This might help in new requests being noticed!

**Comment by mrfloris** [comment]
* Send an email to all players. *
```
> cmi checkcommand mail
[13:51:16 INFO]: --------------------------------------------------
[13:51:16 INFO]: 1. mailall [send/clear/remove] [message]
[13:51:16 INFO]: 2. importfrom [essentials/hd] [home/warp/nick/logoutlocation/money/mail]
[13:51:16 INFO]: 3. mail [send/clear/read] [playerName] (message)
> cmi mailall
[13:51:22 INFO]: Usage: /mailall [send/clear/remove] [message]
[13:51:22 INFO]: * Check or send mail to all players
>
```
Seems to be implemented now.

* Give a reason for AFK * 
```
> cmi checkcommand afk
[13:53:07 INFO]: --------------------------------------------------
[13:53:07 INFO]: 1. afk (-p:playerName) (reason)
[13:53:07 INFO]: 2. afkcheck [playerName]
>
```
This seems to be possible now. You can use the new placeholders inside `config.yml` as well to display their reasons; I use this for example:
```
  AutoAfkCmds:
  - cmi broadcast !&6 * &l[playerDisplayName]&r &ehas gone AFK &7(Automatically)
  # Defines commands to be performed when player enters /cmi afk
  # Supports specialized commands
  ManualAfkCmds:
  - cmi broadcast !&6 * &l[playerDisplayName]&r &ehas gone AFK &7(%cmi_user_afk_msg%)
  # Defines commands to be performed when player leaves afk mode
  AfkLeaveCmds:
  - 'cmi ctellraw all <T>&6 * &l[playerDisplayName] &ehas returned from AFK &7(%cmi_user_afk_msg%)</T><H>&7Has
    been away for: %cmi_user_afk_for%</H>'
```
You can customize these with specialized commands to check `perm:group.admin!` and the other groups, to customize the broadcast msg. 

* Send mail to yourself *
<img width="412" alt="Screenshot 2020-06-16 at 13 56 38" src="https://user-images.githubusercontent.com/28841349/84771164-36c70c80-afd9-11ea-9dae-533519ffc2f1.png">
This seems to be possible now.

* TP Toggle *
```
> cmi checkcommand toggle -p:2
[13:57:31 INFO]: --------------------------------------------------
[13:57:31 INFO]: 11. commandspy (playerName) (-s)
[13:57:31 INFO]: 12. notarget (playerName) (true/false) (-s)
[13:57:31 INFO]: 13. tptoggle (playerName)
[13:57:31 INFO]: 14. flyc (playerName) (true/false) (-s)
[13:57:31 INFO]: 15. signspy (playerName)
[13:57:31 INFO]: 16. togglecompass
[13:57:31 INFO]: For next page perform cmi checkcommand toggle -p:2
> cmi checkcommand tptoggle
[13:57:39 INFO]: --------------------------------------------------
[13:57:39 INFO]: 1. tptoggle (playerName)
> cmi tptoggle
[13:57:43 INFO]: Usage: /tptoggle (playerName)
[13:57:43 INFO]: * Toggle teleportation access to player
>
```
It appears this might be what you were looking for.

Using the same check on commands, it seems clearing inventory, paying money and msgs, all have a toggle. 

* /craft *

You can use cmi aliaseditor to create new craft alias command to open cmi workbench.

* /tpcancel *
```
> cmi checkcommand deny
[14:01:53 INFO]: --------------------------------------------------
[14:01:53 INFO]: 1. tpdeny (playerName)
>
```
This is implemented through tpdeny

* Converting from EssentialsX to CMI - will players lose their inventory *
No, they keep their inventories. 

* Claim kit only once? *
`/cmi kiteditor` will let you edit the kits, from where it can be configured to be limited, unlimited, or a one time. (shift) clicking left/right mouse buttons on the button will increase or decrease the value to the desired value.

There we go, some answers, sorry that it took 2 years, and that it's probably too late. For .. completeness and because i have the answers and this ticket still being open I thought I'd give a courtesy response.

**Comment by Remski01** [comment]
Thanks floris, forget about this. Found after checking the github issue's i opened. (Thoughed it was closed by now.)
